simple-verifier/src/main/java/com/example/simple_verifier/MainActivity.kt

```diff
@@ -0,0 +1,243 @@
+            permissions.add(Manifest.permission.BLUETOOTH_ADVERTISE)
+            permissions.add(Manifest.permission.BLUETOOTH_SCAN)
+            permissions.add(Manifest.permission.BLUETOOTH_CONNECT)
+        }
+
+        return permissions
+    }
+
+    private val permissionsLauncher =
+        registerForActivityResult(ActivityResultContracts.RequestMultiplePermissions()) { permissions ->
+            permissions.entries.forEach {
+                Log.d("permissions", "permissionsLauncher ${it.key} = ${it.value}")
+
+                // Open settings if user denied any required permission
+                if (!it.value && !shouldShowRequestPermissionRationale(it.key)) {
+                    openSettings()
+                    return@registerForActivityResult
+                }
+            }
+        }
+
+    private fun openSettings() {
+        val intent = Intent()
+        intent.action = Settings.ACTION_APPLICATION_DETAILS_SETTINGS
+        intent.data = Uri.fromParts("package", this.applicationContext.packageName, null)
+        startActivity(intent)
+    }
+
+    private fun shouldRequestPermission() {
+        val permissionsNeeded = appPermissions.filter { permission ->
+            ContextCompat.checkSelfPermission(
+                this.applicationContext,
+                permission
+            ) != PackageManager.PERMISSION_GRANTED
+        }
+
+        if (permissionsNeeded.isNotEmpty()) {
+            permissionsLauncher.launch(
+                permissionsNeeded.toTypedArray()
+            )
+        }
+    }
+
+    private fun isAllPermissionsGranted(): Boolean {
+        // If any permission is not granted return false
+        return appPermissions.none { permission ->
+            ContextCompat.checkSelfPermission(
+                this.applicationContext,
+                permission
+            ) != PackageManager.PERMISSION_GRANTED
+        }
+    }
+}
+
+@Composable
+fun LibraryApp(
+    modifier: Modifier = Modifier,
+    supportFragmentManager: FragmentManager,
+    enableReaderPermissions: () -> Unit
+) {
+    Surface(modifier) {
+        LibraryToolboxScreen(modifier = modifier, supportFragmentManager = supportFragmentManager, enableReaderPermissions = enableReaderPermissions)
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun LibraryToolboxScreen(
+    modifier: Modifier = Modifier,
+    supportFragmentManager: FragmentManager,
+    enableReaderPermissions: () -> Unit
+) {
+    Scaffold(
+        topBar = {
+            CenterAlignedTopAppBar(
+                colors = largeTopAppBarColors(
+                    containerColor = MaterialTheme.colorScheme.surfaceContainer,
+                    titleContentColor = MaterialTheme.colorScheme.primary,
+                ),
+                title = {
+                    Text(
+                        style = MaterialTheme.typography.headlineMedium,
+                        text = "Age Assurance Example",
```

Comment:
Please avoid using the term "Age Assurance" as it means a lot of things. Use something like "mDL Age Check Example" instead.